src/pepper/parser.py

```diff
@@ -19,8 +19,26 @@
 import ply.yacc as yacc
 from pepper.lexer import lexer
 from pepper.lexer import tokens  # NOQA
+import pepper.symbol_table as symtable
+from pepper.evaluator import parse_lines, parse_macro
 from pepper.symbol_table import Node
-
+from typing import List, cast
+# precedence according to
+# http://en.cppreference.com/w/c/language/operator_precedence
+
+precedence = (('left', 'BOOL_OR'), ('left', 'BOOL_AND'),
+              ('left', '|'),
+              ('left', '^'),
+              ('left', '&'),
+              ('left', 'COMP_EQU', 'COMP_NEQU'),
+              ('left', '<', '>', 'COMP_LTE', 'COMP_GTE'),
+              ('left', 'L_SHIFT', 'R_SHIFT'),
+              ('left', '+', '-'),
+              ('left', '*', '/', '%'),
+              ('right', '!', '~'))
+
+global if_count
```

Comment:
if possible, and if it doesn't cause too much pain, this should get moved to the symbol table. I try to keep all state in there.

Reply:
moved to symtable